youtube_dl/extractor/atresplayer.py

```diff
@@ -79,124 +64,83 @@ def _login(self):
+            return ExtractorError('{} returned error: {} ({})'.format(
+                self.IE_NAME, data['error'], data.get(
+                    'error_description', 'There is no description')
+            ), expected=True)
+        else:
+            return original_exception
 
     def _real_extract(self, url):
         video_id = self._match_id(url)
 
-        webpage = self._download_webpage(url, video_id)
-
-        episode_id = self._search_regex(
-            r'episode="([^"]+)"', webpage, 'episode id')
-
         request = sanitized_Request(
-            self._PLAYER_URL_TEMPLATE % episode_id,
+            self._PLAYER_URL_TEMPLATE % video_id,
             headers={'User-Agent': self._USER_AGENT})
-        player = self._download_json(request, episode_id, 'Downloading player JSON')
-
-        episode_type = player.get('typeOfEpisode')
-        error_message = self._ERRORS.get(episode_type)
-        if error_message:
-            raise ExtractorError(
-                '%s returned error: %s' % (self.IE_NAME, error_message), expected=True)
+        player = self._download_json(request, video_id,
+                                     'Downloading player JSON')
 
         formats = []
         video_url = player.get('urlVideo')
-        if video_url:
-            format_info = {
-                'url': video_url,
-                'format_id': 'http',
-            }
-            mobj = re.search(r'(?P<bitrate>\d+)K_(?P<width>\d+)x(?P<height>\d+)', video_url)
-            if mobj:
-                format_info.update({
-                    'width': int_or_none(mobj.group('width')),
-                    'height': int_or_none(mobj.group('height')),
-                    'tbr': int_or_none(mobj.group('bitrate')),
-                })
-            formats.append(format_info)
-
-        timestamp = int_or_none(self._download_webpage(
-            self._TIME_API_URL,
-            video_id, 'Downloading timestamp', fatal=False), 1000, time.time())
-        timestamp_shifted = compat_str(timestamp + self._TIMESTAMP_SHIFT)
-        token = hmac.new(
-            self._MAGIC.encode('ascii'),
-            (episode_id + timestamp_shifted).encode('utf-8'), hashlib.md5
-        ).hexdigest()
 
         request = sanitized_Request(
-            self._URL_VIDEO_TEMPLATE.format('windows', episode_id, timestamp_shifted, token),
+            video_url,
             headers={'User-Agent': self._USER_AGENT})
+        try:
+            video_data = self._download_json(request, video_id,
+                                             'Downloading video JSON',
+                                             fatal=True)
+        except ExtractorError as e:
+            if len(e.exc_info) <= 1 or e.exc_info[1].code != 403:
+                raise
+            raise self._atres_player_error(e.exc_info[1].file.read(), e)
+
+        for source in video_data['sources']:
+            if source['type'] == "application/dash+xml":
```

Comment:
Should not break if no `type`.

Reply:
Already fixed on PR bbb857cc948d7ec88398d4b31383df77b69b3a53